spec/models/user_spec.rb

```diff
@@ -2052,16 +2052,44 @@ def hash(password, salt, algorithm = User::TARGET_PASSWORD_ALGORITHM)
     end
 
     describe "#number_of_flagged_posts" do
-      it "counts flagged posts from the user" do
-        Fabricate(:reviewable_flagged_post, target_created_by: user)
+      fab!(:admin) { Fabricate(:admin) }
+
+      it "counts only approved standard flagged posts from the user" do
+        %i[disagree ignore delete_and_ignore].each do |review_action|
+          PostActionCreator
+            .off_topic(admin, Fabricate(:post, user: user))
+            .reviewable
+            .perform(admin, review_action)
+        end
+        PostActionCreator
+          .off_topic(admin, Fabricate(:post, user: user))
+          .reviewable
+          .perform(admin, :agree_and_keep)
+        PostActionCreator
+          .off_topic(admin, Fabricate(:post, user: user))
+          .reviewable
+          .perform(admin, :delete_and_agree)
+        expect(user.number_of_flagged_posts).to eq 2
+      end
 
-        expect(user.number_of_flagged_posts).to eq(1)
+      it "ignores custom flags from the user" do
```

Comment:
I'm wonder if we still need this when this is already being tested in the test case above.

Reply:
The above test case focuses on a standard flag, this test rules out custom flags (`notify_moderators`).

Reply:
Oops sorry I didn't notice that we're calling a different method here :grin: